### ISSUE TITLE
Problem on stop_schedules with duration of multiple days for routes active only on certain days

### ISSUE THREAD

**Issue by eturck** [hove-io/navitia#2279]
Hello!

We are currently working on our stop_schedules API with kraken as the backend. It was pretty simple but while testing I was missing a few departures in my response. I tried using the Navitia API to see if I made a mistake but I have the same problems with it.

One of our line (line 21) is running Monday to Saturday with a different terminus on the week-end.
If I ask, on a Thursday,  the stop_schedules for one of the stop_area on the line, with a duration of two weeks for example (duration=86400*14) I will have : 
- On the week route, all the departures for Thursday and Friday,
- On the week-end route, nothing.
What I think we should have (and what I want :D) : 
- On the week route, all the departures for Thursday and Friday + all next week Monday through Friday + the week after Monday to Wednesday,
- On the week-end route, departures of the next two Saturdays.

I took a quick look at the code and it's pretty clear why it's doing this. In [next_valid_discrete ](https://github.com/CanalTP/navitia/blob/dev/source/routing/next_stop_time.cpp#L118)(in next_stop_time.cpp) :
```cpp
    auto date = DateTimeUtils::date(dt);
    for (const auto* st: dataRaptor.next_stop_time_data.stop_time_range_after(jpp_idx, dt, stop_event)) {
        (...)
    }

    //if none was found, we try again the next day
    date++;
    for (const auto* st: dataRaptor.next_stop_time_data.stop_time_range_forward(jpp_idx, stop_event)) {
        (...)
    }

    // if nothing found, return max
    return {nullptr, DateTimeUtils::inf};
```

We search for the next valid stop time only for today and tomorrow, nothing else.
So in get_stop_times, when [we initialize each journey pattern point with the first departure ](https://github.com/CanalTP/navitia/blob/dev/source/routing/get_stop_times.cpp#L60)we only look for those two days, and same thing [while we loop until max_dt](https://github.com/CanalTP/navitia/blob/dev/source/routing/get_stop_times.cpp#L87) (which his from_datetime + duration), when we are on Friday at the last departure of this day, we only look for the next one Friday and Saturday. Since there is no departure, the next one is Monday, we never output the departures for next week.

As always I have a few questions :) : 
- Is it intended? Since it's in routing I was wondering if with RAPTOR we explicitly don't want to look for a departure  further than the next day.
- If it's not, can we change it? next_valid_discrete can already take a 'bound' to limit its search so I was wondering if I could pas max_dt in it and loop on every day until we are above max_dt?
- It it is intended, what can I do? :cry:

**Comment by TeXitoi** [comment]
There is, in possibly a lot of place, the assumption that we only want something within the next 24h. That's why this method is written like that.

About changing that to search without any hard-coded horizon, it must not generate any problem in journey planing as we precompute all the passages with a fixed horizon.

You may have other problems. It can generate performance regression if no bound is given, but maybe the bounds are always set in the production code. I'm almost sure there is tests that set the bound to infinity. This may or may not cause problem.

So, I'll say, try it, check the unit test, and the performances of departures, arrivals, journeys, stop_schedules and route_schedules, in particular at the border of the data.

**Comment by eturck** [comment]
fixed by #2309
